mobile-app/app/components/TokenIconGroupV2.tsx

```diff
@@ -0,0 +1,80 @@
+import { View } from "@components";
+import { tailwind } from "@tailwind";
+import BigNumber from "bignumber.js";
+import { SymbolIcon } from "@components/SymbolIcon";
+import { translate } from "@translations";
+import { ThemedTextV2 } from "./themed";
+
+interface TokenIconGroupProps {
+  symbols: string[];
+  maxIconToDisplay?: number;
+  testID?: string;
+  size?: number;
+  /**
+   * Flag to "push" token container rightwards, used when no spacing is allowed at the right of TokenIconGroup
+   *
+   * In which the spacing is created by the relative position of each icon to display overlap effect
+   */
+  offsetContainer?: boolean;
+}
+
+export function TokenIconGroupV2({
+  symbols,
+  maxIconToDisplay = 6,
+  testID,
+  size,
+  offsetContainer,
+}: TokenIconGroupProps): JSX.Element {
+  const additionalIcon = BigNumber.max(symbols?.length - maxIconToDisplay, 0);
+  let rightOffset = 0;
+  // increase overlap in case of more than and equal to 5 token icon
+  const overlap = Math.min(symbols.length, maxIconToDisplay) < 5 ? -8 : -12;
```

Comment:
since we are displaying all collaterals, the overlap area should be the same for all 6 icons
```suggestion
  const overlap = Math.min(symbols.length, maxIconToDisplay) < 7 ? -8 : -12;
```

Reply:
on second thought, since we dont have a use case to display different overlap area on v2 UI, let's just do
`const overlap = -8`